--- a/tvheadend-git-22eeadd/configure
+++ b/tvheadend-git-22eeadd/configure
@@ -176,20 +176,6 @@
   COMPILER=gcc
 fi
 
-check_cc_snippet strlcat '#include <string.h>
-int test(int argc, char **argv) {
-  char dst[10];
-  strlcat("test", dst, sizeof(dst));
-  return 0;
-}'
-
-check_cc_snippet strlcpy '#include <string.h>
-int test(int argc, char **argv) {
-  char dst[10];
-  strlcpy("test", dst, sizeof(dst));
-  return 0;
-}'
-
 check_cc_snippet fdatasync '#include <unistd.h>
 int test(int argc, char **argv) {
   fdatasync(0);

